File: Problems/Snakes and Ladders/Golang/player/player.go
package player

type Player struct {
	name            string
	currentPosition int
}

func NewPlayer(name string) *Player {
	return &Player{
		name:            name,
		currentPosition: 0,
	}
}

func (p *Player) GetName() string {
	return p.name
}

func (p *Player) GetCurrentPosition() int {
	return p.currentPosition
}

func (p *Player) SetCurrentPosition(pos int) {
	p.currentPosition = pos
}

File: Problems/Snakes and Ladders/Java/player/Player.java
package player;

public class Player {
    private String name;
    private int currentPosition;

    public Player(String name) {
        this.name = name;
        this.currentPosition = 0; // Starting position
    }

    public String getName() {
        return name;
    }

    public int getCurrentPosition() {
        return currentPosition;
    }

    public void setCurrentPosition(int currentPosition) {
        this.currentPosition = currentPosition;
    }
}
